plugins/talk-plugin-slack-notifications/server/hooks.js

```diff
@@ -19,24 +19,26 @@ module.exports = {
           }
         } = result;
         const username = context.user.username;
-        const response = await fetch(SLACK_WEBHOOK_URL, {
-          method: 'POST',
-          headers: {
-            'Content-Type': 'application/json',
-          },
-          timeout: SLACK_WEBHOOK_TIMEOUT,
-          body: JSON.stringify({
-            attachments: [{
-              text: text,
-              footer: `Comment by ${username}`,
-              ts: Math.floor(Date.parse(createdAt) / 1000),
-            }]
-          }),
+        process.nextTick(async () => {
```

Comment:
I thought it would be better to keep the field extraction from `result` and `context` outside the nextTick callback to avoid holding references to those objects for longer than the lifetime of the request - but I'm new to this stuff so comments are welcome :smile:

Reply:
You can certainly extract those outside to the function and access them from the closure. It's up to you!